pennylane_forest/qvm.py

```diff
@@ -170,21 +170,74 @@ def pre_expval(self):
             self.state[q] = bitstring_array[:, i]
 
     def expval(self, expectation, wires, par):
-        evZ = np.mean(1-2*self.state[wires[0]])
+        if len(wires) == 1:
+            # 1 qubit observable
+            evZ = np.mean(1-2*self.state[wires[0]])
 
-        # for single qubit state probabilities |psi|^2 = (p0, p1),
-        # we know that p0+p1=1 and that <Z>=p0-p1
-        p0 = (1+evZ)/2
-        p1 = (1-evZ)/2
+            # for single qubit state probabilities |psi|^2 = (p0, p1),
+            # we know that p0+p1=1 and that <Z>=p0-p1
+            p0 = (1+evZ)/2
+            p1 = (1-evZ)/2
 
-        if expectation == 'Identity':
-            # <I> = \sum_i p_i
-            return p0+p1
+            if expectation == 'Identity':
+                # <I> = \sum_i p_i
+                return p0+p1
```

Comment:
Minor quibble, but this and other similar expressions might be more readable if there are spaces between the operands and binary operator, so e.g. `p0+p1` -> `p0 + p1` etc.

Reply:
You're right, this is also specified in PEP8 :+1:

Reply:
There are a few other spots where we could make a similar change: `p0 = (1+evZ)/2` and `p1 = (1-evZ)/2` (lines 179-180), `return w[0]*p0 + w[1]*p1` -> `return w[0] * p0 + w[1] * p1` (line 190), `cb = np.sum(2**np.arange(len(wires)-1, -1, -1)*i)` -> `cb = np.sum(2**np.arange(len(wires)-1, -1, -1) * i)` (line 233) all in `qvm.py`.

In `numpy_wavefunction.py`: `ev = (n0*a[0] +(self.shots-n0)*a[1]) / self.shots` -> `ev = (n0 * a[0] +(self.shots - n0) * a[1]) / self.shots` (line 80), `inactive_wires = list(set(range(N))-set(wires))` -> `inactive_wires = list(set(range(N)) - set(wires))` (line 126), `rearranged_wires = np.array(list(wires)+inactive_wires)` -> `rearranged_wires = np.array(list(wires) + inactive_wires)` (line 132), `perm = np.ravel_multi_index(tuples[:, rearranged_wires].T, [2]*N)` -> `perm = np.ravel_multi_index(tuples[:, rearranged_wires].T, [2] * N)` (line 138).

Reply:
What I might do is apply the [black](https://github.com/python/black) Python formatter to all source codes - I'll do that in a separate PR, so as not to mess with the diff too much.

Reply:
Sounds good to me!